streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamsProducerTest.java

```diff
@@ -688,12 +793,55 @@ public void shouldThrowStreamsExceptionOnEosAbortTxError() {
 
     @Test
     public void shouldFailOnEosAbortTxFatal() {
-        eosMockProducer.abortTransactionException = new RuntimeException("KABOOM!");
+        eosAlphaMockProducer.abortTransactionException = new RuntimeException("KABOOM!");
         // call `send()` to start a transaction
-        eosStreamsProducer.send(record, null);
+        eosAlphaStreamsProducer.send(record, null);
 
-        final RuntimeException thrown = assertThrows(RuntimeException.class, eosStreamsProducer::abortTransaction);
+        final RuntimeException thrown = assertThrows(RuntimeException.class, eosAlphaStreamsProducer::abortTransaction);
 
         assertThat(thrown.getMessage(), is("KABOOM!"));
     }
+
+
+    // EOS beta test
+
+    // functional tests
+
+    @Test
+    public void shouldSetSetNewProducerOnResetProducer() {
+        final Producer<byte[], byte[]> producer = mock(Producer.class);
+
+        eosBetaStreamsProducer.resetProducer(producer);
+
+        assertThat(eosBetaStreamsProducer.kafkaProducer(), is(producer));
+    }
+
+    @Test
+    public void shouldResetTransactionInitializedOnResetProducer() {
+        final Producer<byte[], byte[]> producer = mock(Producer.class);
+
+        eosBetaStreamsProducer.initTransaction();
+        eosBetaStreamsProducer.resetProducer(producer);
+
+        producer.initTransactions();
```

Comment:
If the `StreamsProducer` flag would not be reset, this call would not be made below in `eosBetaStreamsProducer.initTransaction();`